server/util/flagutil/types/autoflags/tags/tags.go

```diff
@@ -0,0 +1,199 @@
+package tags
+
+import (
+	"flag"
+	"fmt"
+
+	"github.com/buildbuddy-io/buildbuddy/server/util/flagutil/common"
+	"github.com/buildbuddy-io/buildbuddy/server/util/log"
+)
+
+type Tagged interface {
+	Value() flag.Value
+	Zero() string
+
+	// from flag.Value
+	SetFunc(func(value string) error)
+	StringFunc(func() string)
+
+	// from WrappingValue
+	WrappedValueFunc(func() flag.Value)
+
+	// from Expandable
+	ExpandFunc(func(mapping func(string) (string, error)) error)
+
+	// from Secretable
+	IsSecretFunc(func() bool)
+
+	// from SetValueForFlagNameHooked
+	SetValueForFlagNameHookFunc(func())
+
+	// from YAMLSetValueHooked
+	YAMLSetValueHookFunc(func())
+}
+
+type Taggable interface {
+	Tag(flagset *flag.FlagSet, name string, f Tagged)
+}
+
+type TagFlag[T any] struct {
```

Comment:
could you add a comment explaining what this is? The name doesn't feel very intuitive to me. Would `TaggedFlag` be more accurate?